edgecast/routedns/group.go

```diff
@@ -62,7 +64,47 @@ func (svc *RouteDNSService) AddGroup(params AddGroupParams) (*int, error) {
 			err,
 		)
 	}
-	return &groupID, nil
+
+	if groupID == -1 {
+		return nil, fmt.Errorf(
+			"AddGroup->Group creation failed. Group ID == -1. Please try again",
```

Comment:
If a function can possibly return multiple errors it may be good to create a typed package error (ie. `var ErrAddGroupCreation = errors.New("..")`) so that a user can handle the error depending on the type.

Reply:
@defstream this is a great suggestion, thank you. I will bring this up in the tech meeting for us to discuss as a team. This is a good example where the user might want to do something different in this case (retry). I suggest we proceed with this change as is for consistency with the rest of the SDK but we should identify how we want to handle errors overall and create a ticket to update all functions in the SDK.